pkg/loki/curator/config/config.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"io/ioutil"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+// CuratorConfig holds the curator's configurations
+type CuratorConfig struct {
+	LogLevel        string        `yaml:"LogLevel,omitempty"`
+	DiskPath        string        `yaml:"DiskPath,omitempty"`
+	TriggerInterval time.Duration `yaml:"TriggerInterval,omitempty"`
+	InodeConfig     Config        `yaml:"InodeConfig,omitempty"`
+	StorageConfig   Config        `yaml:"StorageConfig,omitempty"`
+}
+
+// Config holds the curator's config for a unit
+type Config struct {
+	MinFreePercentages             int `yaml:"MinFreePercentages,omitempty"`
+	TargetFreePercentages          int `yaml:"TargetFreePercentages,omitempty"`
+	PageSizeForDeletionPercentages int `yaml:"PageSizeForDeletionPercentages,omitempty"`
```

Comment:
Hm, how is `omitempty` working with `int`?
It is better to use `*int` which allow to properly differ between unset value `nil` and zero value `0`.

On the other side, I see these values are required, thus you can make them mandatory and keep them as `int`.

Reply:
omitempty for int will check if the given field is not in its default state (e.g. 0 for int).